docs/docs/30-how-to-guides/14-working-with-stages.md

```diff
@@ -0,0 +1,556 @@
+:::
+
+In the following example, the `test` `Stage` requests `Freight` that has
+originated from the `my-warehouse` `Warehouse` and indicates that it will accept
+new `Freight` _directly_ from that origin:
+
+```yaml
+apiVersion: kargo.akuity.io/v1alpha1
+kind: Stage
+metadata:
+  name: test
+  namespace: kargo-demo
+spec:
+  requestedFreight:
+  - origin:
+      kind: Warehouse
+      name: my-warehouse
+    sources:
+      direct: true
+  # ...
+# ...
+```
+
+In this example, the `uat` `Stage` requests `Freight` that has originated from
+the `my-warehouse` `Warehouse`, but indicates that it will accept such `Freight`
+only after it has been _verified_ in the `test` `Stage`:
+
+```yaml
+apiVersion: kargo.akuity.io/v1alpha1
+kind: Stage
+metadata:
+  name: uat
+  namespace: kargo-demo
+spec:
+  requestedFreight:
+  - origin:
+      kind: Warehouse
+      name: my-warehouse
+    sources:
+      stages:
+      - test
+  # ...
+```
+
+Stages may also request `Freight` from multiple sources. The following example
+illustrates a `Stage` that requests `Freight` from both a `microservice-a` and
+`microservice-b` `Warehouse`:
+
+```yaml
+apiVersion: kargo.akuity.io/v1alpha1
+kind: Stage
+metadata:
+  name: test
+  namespace: kargo-demo
+spec:
+  requestedFreight:
+  - origin:
+      kind: Warehouse
+      name: microservice-a
+    sources:
+      direct: true
+  - origin:
+      kind: Warehouse
+      name: microservice-b
+    sources:
+      direct: true
+  # ...
+```
+
+:::tip
+By requesting `Freight` from multiple sources, a `Stage` can effectively
+participate in _multiple pipelines_ that may each deliver different collections
+of artifacts independently of the others.
+:::
+
+### Promotion Templates
+
+The `spec.promotionTemplate` field is used to describe _how_ to transition
+`Freight` into the `Stage`. The `spec.promotionTemplate.steps` field describes
+the discrete steps of a promotion process in detail.
+
+In the following, very common example, the `promotionTemplate` describes steps
+to:
+
+1. Clone a Git repository containing Kubernetes manifests and Kustomize
+configuration, checking out two different branches to two different
+directories.
+
+1. Clears the contents of one working tree, with intentions to fully replace its
+contents.
+
+1. Runs the equivalent of `kustomize edit set image` to update a
+`kustomization.yaml` file with a reference to an updated
+`public.ecr.aws/nginx/nginx` container image.
+
+1. Renders the updated manifests using the equivalent of `kustomize build`.
+
+1. Commits the updated manifests and pushes them to the `stage/test` of the
+remote repository.
+
+1. Forces Argo CD to sync the `kargo-demo-test` application to the latest commit
+of the `stage/test` branch.
+
+```yaml
+promotionTemplate:
+  spec:
+    steps:
+    - uses: git-clone
+      config:
+        repoURL: https://github.com/example/repo.git
+        checkout:
+        - fromFreight: true
```

Comment:
This example needs to be updated to favor the use of expressions over deprecated fields such as `fromFreight`. Look at the promotion steps reference or my own examples repo for guidance on what a more up-to-date version of this process looks like.